### ISSUE TITLE
Dubbo 3.0 with mybatis create a NullPointerException

### ISSUE THREAD

**Issue by DUANKD** [apache/dubbo#8127]
- [x] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.
- [x] I have checked the [FAQ](https://github.com/apache/dubbo/blob/master/FAQ.md) of this repository and believe that this is not a duplicate.

### Environment

* Dubbo version: 3.0.0
* Operating System version: windows 10
* Java version: OpenJDK 64-Bit Server VM AdoptOpenJDK-11.0.11+9 
* spring-boot-starter-parent : 2.3.0.RELEASE
* Mybatis version: 3.5.6
* com.baomidou : mybatis-plus-boot-starter : 3.4.2

### Steps to reproduce this issue

1. maven build dubbo-spring-boot-zookeeper-sample application 
2. I get error messages when I start run，It may be caused by Dubbo autoconfig or org.apache.ibatis.annotations.Mapper;
3. Test dubbo version 2.7 can run normally

[code address](https://github.com/DUANKD/dubbo-test) is here.

### Expected Result

run success

### Actual Result

run failure ,When parsing a custom mapper, an npe exception is generated.
The reason is annotatedBeanDefinition.getFactoryMethodMetadata() is null.Please check it.
There is the exception trace:

```
2021-06-24 10:09:57.383 ERROR 30372 --- [           main] o.s.boot.SpringApplication               : Application run failed

java.lang.NullPointerException: null
	at org.apache.dubbo.config.spring.beans.factory.annotation.ReferenceAnnotationBeanPostProcessor.isAnnotatedReferenceBean(ReferenceAnnotationBeanPostProcessor.java:165) ~[dubbo-3.0.0.jar:3.0.0]
	at org.apache.dubbo.config.spring.beans.factory.annotation.ReferenceAnnotationBeanPostProcessor.postProcessBeanFactory(ReferenceAnnotationBeanPostProcessor.java:131) ~[dubbo-3.0.0.jar:3.0.0]
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:291) ~[spring-context-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.context.support.PostProcessorRegistrationDelegate.invokeBeanFactoryPostProcessors(PostProcessorRegistrationDelegate.java:175) ~[spring-context-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.invokeBeanFactoryPostProcessors(AbstractApplicationContext.java:706) ~[spring-context-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:532) ~[spring-context-5.2.6.RELEASE.jar:5.2.6.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:758) ~[spring-boot-2.3.0.RELEASE.jar:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:750) ~[spring-boot-2.3.0.RELEASE.jar:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:397) ~[spring-boot-2.3.0.RELEASE.jar:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:315) ~[spring-boot-2.3.0.RELEASE.jar:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1237) ~[spring-boot-2.3.0.RELEASE.jar:2.3.0.RELEASE]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226) ~[spring-boot-2.3.0.RELEASE.jar:2.3.0.RELEASE]
	at org.example.dubbo_test.DubboProviderDemo.main(DubboProviderDemo.java:18) ~[classes/:na]
```

**Comment by pinxiong** [comment]
I'll take a look at this issue

**Comment by pinxiong** [comment]
This issue has fixed, you can refer to #8100 and #8080

**Comment by xiaofu1113** [comment]
> 这个问题已经修复，可以参考#8100和#8080
是合并到3.0上去吗？我拉去的源代码，还是一样的！！！问题没有修复呀？

**Comment by pinxiong** [comment]
> > 这个问题已经修复，可以参考#8100和#8080
> 
> 是合并到3.0上去吗？我拉去的源代码，还是一样的！！！问题没有修复呀？

Yes, it has been merged in branch [3.0](https://github.com/apache/dubbo/commits/3.0). You can see it [here](https://github.com/apache/dubbo/blob/3.0/dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ReferenceAnnotationBeanPostProcessor.java#L162)

**Comment by xiaofu1113** [comment]
> > > 这个问题已经修复，可以参考#8100和#8080
> > 
> > 
> > 是合并到3.0上去吗？我拉去的源代码，还是一样的！！！问题没有修复呀？
> 
> Yes, it has been merged in branch [3.0](https://github.com/apache/dubbo/commits/3.0). You can see it [here](https://github.com/apache/dubbo/blob/3.0/dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ReferenceAnnotationBeanPostProcessor.java#L162)
仓库的包也更新一下呗

**Comment by 15168326318** [comment]
> > > > 这个问题已经修复，可以参考#8100和#8080
> > > 
> > > 
> > > 是合并到3.0上去吗？我拉去的源代码，还是一样的！！！问题没有修复呀？
> > 
> > 
> > Yes, it has been merged in branch [3.0](https://github.com/apache/dubbo/commits/3.0). You can see it [here](https://github.com/apache/dubbo/blob/3.0/dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ReferenceAnnotationBeanPostProcessor.java#L162)
> > 仓库的包也更新一下呗

我jar删除了，重新拉了也不行啊。是不是公网仓库没更新呀。

**Comment by 15168326318** [comment]
是  dubbo  3.0.1版本修复了 ？

**Comment by pinxiong** [comment]
> 是 dubbo 3.0.1版本修复了 ？

Yes! You can see release info in [3.0.1](https://github.com/apache/dubbo/releases/tag/dubbo-3.0.1)

**Comment by pinxiong** [comment]
> > > > 这个问题已经修复，可以参考#8100和#8080
> > > 
> > > 
> > > 是合并到3.0上去吗？我拉去的源代码，还是一样的！！！问题没有修复呀？
> > 
> > 
> > Yes, it has been merged in branch [3.0](https://github.com/apache/dubbo/commits/3.0). You can see it [here](https://github.com/apache/dubbo/blob/3.0/dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ReferenceAnnotationBeanPostProcessor.java#L162)
> > 仓库的包也更新一下呗

You can see release info in [3.0.1](https://github.com/apache/dubbo/releases/tag/dubbo-3.0.1)

**Comment by xiaofu1113** [comment]
> > > > > 这个问题已经修复，可以参考#8100和#8080
> > > > 
> > > > 
> > > > 是合并到3.0上去吗？我拉去的源代码，还是一样的！！！问题没有修复呀？
> > > 
> > > 
> > > Yes, it has been merged in branch [3.0](https://github.com/apache/dubbo/commits/3.0). You can see it [here](https://github.com/apache/dubbo/blob/3.0/dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ReferenceAnnotationBeanPostProcessor.java#L162)
> > > 仓库的包也更新一下呗
> 
> You can see release info in [3.0.1](https://github.com/apache/dubbo/releases/tag/dubbo-3.0.1)

maven中央仓库最新版本是3.0.0

**Comment by 15168326318** [comment]
> maven中央仓库最新版本是3.0.0

我今天拉了maven中央仓库版本3.0.0的包，可还是不行。   修改的代码没合并过来，还是老的代码。

**Comment by 15168326318** [comment]
> > > > > 这个问题已经修复，可以参考#8100和#8080
> > > > 
> > > > 
> > > > 是合并到3.0上去吗？我拉去的源代码，还是一样的！！！问题没有修复呀？
> > > 
> > > 
> > > Yes, it has been merged in branch [3.0](https://github.com/apache/dubbo/commits/3.0). You can see it [here](https://github.com/apache/dubbo/blob/3.0/dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ReferenceAnnotationBeanPostProcessor.java#L162)
> > > 仓库的包也更新一下呗
> 
> You can see release info in [3.0.1](https://github.com/apache/dubbo/releases/tag/dubbo-3.0.1)

问题是3.0.1的版本还没发布到maven中央仓库啊

**Comment by pinxiong** [comment]
> > > > > > 这个问题已经修复，可以参考#8100和#8080
> > > > > 
> > > > > 
> > > > > 是合并到3.0上去吗？我拉去的源代码，还是一样的！！！问题没有修复呀？
> > > > 
> > > > 
> > > > Yes, it has been merged in branch [3.0](https://github.com/apache/dubbo/commits/3.0). You can see it [here](https://github.com/apache/dubbo/blob/3.0/dubbo-config/dubbo-config-spring/src/main/java/org/apache/dubbo/config/spring/beans/factory/annotation/ReferenceAnnotationBeanPostProcessor.java#L162)
> > > > 仓库的包也更新一下呗
> > 
> > 
> > You can see release info in [3.0.1](https://github.com/apache/dubbo/releases/tag/dubbo-3.0.1)
> 
> 问题是3.0.1的版本还没发布到maven中央仓库啊

It's being voted on and will be released this weekend.
